transformer/compose/utils.go

```diff
@@ -58,6 +72,257 @@ func IsV3(path string) (bool, error) {
 }
 */
 
+func createConfigMapName(sourcePath string) string {
+	cfgMapHashID := getHash([]byte(sourcePath))
+	cfgName := fmt.Sprintf("%s-%d", cfgMapPrefix, cfgMapHashID)
+	cfgName = common.MakeStringK8sServiceNameCompliant(cfgName)
+	return cfgName
+}
+
+func createSecretName(sourcePath string) string {
+	cfgMapHashID := getHash([]byte(sourcePath))
+	cfgName := fmt.Sprintf("%s-%d", secretPrefix, cfgMapHashID)
+	cfgName = common.MakeStringK8sServiceNameCompliant(cfgName)
+	return cfgName
+}
+
+func createVolumeName(sourcePath string, serviceName string) string {
+	hashID := getHash([]byte(sourcePath))
+	volumeName := fmt.Sprintf("%s-%s-%d", common.VolumePrefix, serviceName, hashID)
+	volumeName = common.MakeStringK8sServiceNameCompliant(volumeName)
+	return volumeName
+}
+
+func withinSizeLimit(filePath string) (bool, error) {
+	fileInfo, err := os.Stat(filePath)
+	if err != nil {
+		return false, err
+	}
+
+	if !fileInfo.IsDir() {
+		if fileInfo.Size() > int64(maxConfigMapSizeLimit) {
+			return false, fmt.Errorf("config map could not be created from file. Size limit of 1M exceeded")
+		}
+	} else {
+		var totalSize int64 = 0
+		filepath.Walk(filePath, func(dir string, fileHandle os.FileInfo, err error) error {
+			if !fileHandle.IsDir() {
+				totalSize += fileHandle.Size()
```

Comment:
We need not calculate the size of the entire folder if the size is greater than the size limit. We can move the check here, so that we break out as soon as the size reaches 1 MB?

Reply:
Done